CondFormats/SiStripObjects/src/SiStripThreshold.cc

```diff
@@ -143,19 +143,14 @@ void SiStripThreshold::printSummary(std::stringstream& ss, const TrackerTopology
     minHth = 10000;
     minCth = 10000;  //min value
     n = 0;
-    firstStrip = 0;
     for (; it != eit; ++it) {
       itp = it + 1;
       firstStrip = it->getFirstStrip();
       if (itp != eit)
         stripRange = (itp->getFirstStrip() - firstStrip);
       else
-        stripRange =
-            firstStrip > 511
-                ? 768 - firstStrip
-                : 512 -
-                      firstStrip;  //*FIXME, I dont' know ithis class the strip number of a detector, so I assume wrongly that if the last firstStrip<511 the detector has only 512 strips. Clearly wrong. to be fixed
-
+        //*FIXME, I dont' know ithis class the strip number of a detector, so I assume wrongly that if the last firstStrip<511 the detector has only 512 strips. Clearly wrong. to be fixed
+        stripRange = firstStrip > 511 ? 768 - firstStrip : 512 - firstStrip;
```

Comment:
Just for the record, I *think* one could make this exact by means of using the `StripTopology` (you can get it from this method

https://github.com/cms-sw/cmssw/blob/6d26b89d99a92507aad02623bd3feac40f65d369/Geometry/TrackerGeometryBuilder/interface/StripGeomDetUnit.h#L41
)
the con is that one would need to pass a `TrackerGeometry` in input to `SiStripThreshold::printSummary`.
An alternative way is to use the mechanism from

https://github.com/cms-sw/cmssw/blob/6d26b89d99a92507aad02623bd3feac40f65d369/CalibFormats/SiStripObjects/interface/SiStripDetInfo.h#L55

and use the fact that

```C++
unsigned int nStrip = detInfo_.getNumberOfApvsAndStripLength(detid).first * sistrip::STRIPS_PER_APV;
```